app/services/api_user_token_service.rb

```diff
@@ -0,0 +1,77 @@
+class ApiUserTokenService
+  def initialize(config = nil, args = {})
+    @config = config || YAML.load_file(Rails.root.join("config/api.yml"))
+    @svc_options = args
+    token_mgr
+  end
+
+  # Additional Requester type token ttl's for authentication
+  #
+  REQUESTER_TTL_CONFIG = {"ui" => :ui_token_ttl}.freeze
+
+  # API Settings with additional token ttl's
+  #
+  def api_config
+    @api_config ||= Settings[base_config[:module]].to_hash.merge(
+      REQUESTER_TTL_CONFIG["ui"] => Settings.session.timeout
+    )
+  end
+
+  def token_mgr
```

Comment:
Since this is being called by the constructor, WDYT about putting this logic there and making this an `attr_reader`?